src/NewtonKrylov.fypp

```diff
@@ -0,0 +1,418 @@
+         !! Target tolerance
+         real(${kind}$), intent(in)  :: rnorm
+         !! Norm of the residual of the current iterate
+         integer,  intent(in)  :: iter
+         !! Newton iteration count
+         integer,  intent(out)  :: info
+         !! Information flag
+      end subroutine abstract_scheduler_${kind}$
+
+      #:endfor
+   end interface
+
+contains
+
+   #:for kind, type in RC_KINDS_TYPES
+   subroutine newton_${type[0]}$${kind}$(sys, X, info, tolerance, options, linear_solver, linear_solver_options, preconditioner, scheduler)
+      class(abstract_system_${type[0]}$${kind}$),                         intent(inout) :: sys
+      !! Dynamical system for which we wish to compute a fixed point
+      class(abstract_vector_${type[0]}$${kind}$),                         intent(inout) :: X
+      !! Initial guess for the fixed point, will be overwritten with solution
+      integer,                                            intent(out)   :: info
+      !! Information flag
+      real(${kind}$),                                 optional, intent(in)    :: tolerance
+      real(${kind}$)                                                          :: target_tol
+      !! Target absolute solver tolerance
+      type(newton_${kind}$_opts),                     optional, intent(in)    :: options
+      type(newton_${kind}$_opts)                                              :: opts
+      !! Options for the Newton-Krylov iteration
+      procedure(abstract_linear_solver_${type[0]}$${kind}$),    optional                :: linear_solver
+      !! Linear solver to be used to find Newton step
+      class(abstract_opts),                     optional, intent(in)    :: linear_solver_options
+      class(abstract_opts), allocatable                                 :: solver_opts
+      !! Options for the linear solver
+      class(abstract_precond_${type[0]}$${kind}$),              optional, intent(in)    :: preconditioner
+      class(abstract_precond_${type[0]}$${kind}$), allocatable                          :: precond
+      !! Preconditioner for the linear solver
+      procedure(abstract_scheduler_${kind}$),         optional                :: scheduler
+
+      !--------------------------------------
+      !-----     Internal variables     -----
+      !--------------------------------------
+      
+      class(abstract_vector_${type[0]}$${kind}$), allocatable :: residual, increment
+      real(${kind}$) :: rnorm, tol
+      logical :: converged, has_precond, has_solver_opts
+      integer :: i, maxiter, maxstep_bisection
+      character(len=256) :: msg
+      procedure(abstract_linear_solver_${type[0]}$${kind}$), pointer :: solver => null()
+      procedure(abstract_scheduler_${kind}$), pointer :: tolerance_scheduler => null()
+
+      ! Newton-solver tolerance
+      if (present(tolerance)) then
+         target_tol = tolerance
+      else
+         target_tol = atol_${kind}$
+      end if
```

Comment:
`stdlib_optval`?

Reply:
yup, for this option it's actually possible.